ecl/hthor/hthor.cpp

```diff
@@ -8300,24 +8301,29 @@ bool CHThorDiskReadBaseActivity::openNext()
             /* If part can potentially be remotely streamed, 1st check if any part is local,
              * then try to remote stream, and otherwise failover to legacy remote access
              */
-            unsigned localCopy = NotFound;
+            unsigned startCopy = 0;
             if (tryRemoteStream && (rt_binary == readType))
             {
                 std::vector<unsigned> remoteCandidates;
                 // scan for local part 1st
                 for (unsigned copy=0; copy<numCopies; copy++)
                 {
                     RemoteFilename rfn;
-                    curPart->getFilename(rfn, copy);
-                    StringBuffer localPath;
-                    if (!isRemoteReadCandidate(agent, rfn, localPath))
+                    if (curPart)
+                        curPart->getFilename(rfn,copy);
+                    else
+                        ldFile->getPartFilename(rfn, partNum, copy);
+
+                    if (!isRemoteReadCandidate(agent, rfn))
                     {
-                        Owned<IFile> iFile = createIFile(localPath.str());
+                        StringBuffer path;
+                        rfn.getPath(path);
```

Comment:
Should this be getLocalPath()?

Reply:
No that was a deliberate change.
In 99% cases in this context getPath() will call getLocalPath() anyway,
but if forceRemoteDisabled is set, then the location could be remote, and so this could should get the full URL, which getPath() will